stacks_queues/stack.py

```diff
@@ -21,18 +25,25 @@ def pop(self):
             The Stack is empty.
             returns None
         """
-        pass
+        if self.store.length == 0:
+            raise StackEmptyException ("stack is empty")
+
+        return self.store.remove_first()
 
     def empty(self):
         """ Returns True if the Stack is empty
             And False otherwise
         """
-        pass
+        self.store.length = 0
+        if self.store.length == 0:
+            return True
+        return False
 
     def __str__(self):
         """ Returns the Stack in String form like:
             [3, 4, 7]
             Starting with the top of the Stack and
             ending with the bottom of the Stack.
         """
-        pass
+        return self.store.__str__
```

Comment:
The`__str__` method is an example of what we call operator overloading. The `str` method is built-in for all data types (even classes you create) in Python. When we want the `str` method to do something different than the default built-in behavior, we can 'overload' the function by implementing new behavior under the `__str__` method. However when we call the function, we still use `str` without the dunders.

You can also do this with other built in operators and functions. For example, you can override the behavior of the `==` operator with `__eq__`

```suggestion
        return str(self.store)
```